numba_cuda/numba/cuda/tests/nrt/test_nrt.py

```diff
@@ -100,11 +73,157 @@ def g(out_ary):
 
         out_ary = np.zeros(1, dtype=np.int64)
 
-        with patch('numba.config.CUDA_ENABLE_NRT', True, create=True):
-            g[1,1](out_ary)
+        g[1,1](out_ary)
 
         self.assertEqual(out_ary[0], 1)
 
 
+class TestNrtStatistics(CUDATestCase):
+
+    def setUp(self):
+        self._stream = cuda.default_stream()
+        # Store the current stats state
+        self.__stats_state = rtsys.memsys_stats_enabled(self._stream)
+
+    def tearDown(self):
+        # Set stats state back to whatever it was before the test ran
+        if self.__stats_state:
+            rtsys.memsys_enable_stats(self._stream)
+        else:
+            rtsys.memsys_disable_stats(self._stream)
+
+    def test_stats_env_var_explicit_on(self):
+        # Checks that explicitly turning the stats on via the env var works.
+        src = """if 1:
+        from numba import cuda
+        from unittest.mock import patch
+        from numba.cuda.runtime import rtsys
+        from numba.cuda.tests.nrt.mock_numpy import cuda_arange
+
+        @cuda.jit
+        def foo():
+            x = cuda_arange(10)[0]
+
+        # initialize the NRT before use
+        rtsys.initialize()
```

Comment:
I haven't got this test working locally yet (or understood it fully) but it seems we need to allocate before initialization:

```suggestion
        rtsys.allocate()
        rtsys.initialize()
```